lib/middleware/tagged_logging.rb

```diff
@@ -20,5 +22,14 @@ def call(env)
       ::Logging.mdc.delete('request')
       ::Logging.mdc.delete('session')
     end
+
+    private
+
+    # Prefer using a digest of the session ID so it can be passed safely to external smart proxies
+    # etc to prevent session hijacks
+    def digest_id(id)
+      salt = defined?(EncryptionKey::ENCRYPTION_KEY) ? EncryptionKey::ENCRYPTION_KEY : 'default'
+      OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha256'), salt, id)
+    end
```

Comment:
On the other hand, this may weaken our ENCRYPTION_KEY. I am no security expert but it's possible, imagine id is empty or stolen already.

Anyway, make it public and class method, we will need it in order to find matching records coming from proxy via log buffer API.